src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -73,10 +73,18 @@ extends AbstractMap[A, B]
   }
 
   override def getOrElseUpdate(key: A, defaultValue: => B): B = {
-    val i = index(elemHashCode(key))
+    val hash = elemHashCode(key)
+    val table0 = table
```

Comment:
This is too early.  You don't have to store table0 until right before the `default = defaultValue` inside the `else`.